fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/BitmapAggregateIndexExpansionVisitor.java

```diff
@@ -0,0 +1,126 @@
+import com.apple.foundationdb.record.query.plan.cascades.values.ArithmeticValue;
+import com.apple.foundationdb.record.query.plan.cascades.values.FieldValue;
+import com.apple.foundationdb.record.query.plan.cascades.values.FunctionCatalog;
+import com.apple.foundationdb.record.query.plan.cascades.values.LiteralValue;
+import com.apple.foundationdb.record.query.plan.cascades.values.NumericAggregationValue;
+import com.apple.foundationdb.record.query.plan.cascades.values.RecordConstructorValue;
+import com.apple.foundationdb.record.query.plan.cascades.values.Value;
+import com.apple.foundationdb.record.util.pair.NonnullPair;
+import com.google.common.base.Verify;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+
+import javax.annotation.Nonnull;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * The bitmap aggregate index expansion visitor.
+ */
+public class BitmapAggregateIndexExpansionVisitor extends AggregateIndexExpansionVisitor {
+    /**
+     * Constructs a new instance of {@link BitmapAggregateIndexExpansionVisitor}.
+     *
+     * @param index The target index.
+     * @param recordTypes The indexed record types.
+     */
+    public BitmapAggregateIndexExpansionVisitor(@Nonnull final Index index, @Nonnull final Collection<RecordType> recordTypes) {
+        super(index, recordTypes);
+        Verify.verify(IndexTypes.BITMAP_VALUE.equals(index.getType()));
+    }
+
+    @Nonnull
+    @Override
+    protected NonnullPair<Quantifier, List<Placeholder>> constructGroupBy(@Nonnull final Quantifier selectWhereQun, @Nonnull final GraphExpansion baseExpansion) {
+        if (groupingKeyExpression.getGroupedCount() != 1) {
+            throw new UnsupportedOperationException("bitmap aggregate index is expected to contain exactly one grouped expression, however it contains " + groupingKeyExpression.getGroupedCount() + " aggregations");
+        }
+        final Value groupedValue = baseExpansion.getResultColumns().get(groupingKeyExpression.getGroupingCount()).getValue();
+
+        // construct aggregation RCV
+        final Value argument;
+        if (groupedValue instanceof FieldValue) {
+            final var aliasMap = AliasMap.identitiesFor(Sets.union(selectWhereQun.getCorrelatedTo(),
+                    groupedValue.getCorrelatedTo()));
+            final var result = selectWhereQun.getRangesOver().get().getResultValue()
+                    .pullUp(List.of(groupedValue), aliasMap, ImmutableSet.of(), selectWhereQun.getAlias());
+            if (!result.containsKey(groupedValue)) {
+                throw new RecordCoreException("could not pull grouped value " + groupedValue)
+                        .addLogInfo(LogMessageKeys.VALUE, groupedValue);
+            }
+            argument = result.get(groupedValue);
+        } else {
+            throw new RecordCoreException("unable to plan group by with non-field value")
```

Comment:
If the previous exception is `UnsupportedOperationException` I would say this one should be as well.